config/fi_provider.m4

```diff
@@ -127,8 +127,15 @@ AC_DEFUN([FI_PROVIDER_SETUP],[
 		 AS_IF([test -f "$srcdir/prov/$1/include/rdma/fi_direct.h"],
 			[AC_MSG_RESULT(yes)],
 			[AC_MSG_RESULT(no)
-			  AC_MSG_ERROR([$1 provider was requested as direct, but is missing fi_direct.h])]
-			)])
+			 AC_MSG_CHECKING(for $srcdir/prov/$1/include/rdma/fi_direct.h.in)
+			 AS_IF([test -f "$srcdir/prov/$1/include/rdma/fi_direct.h.in"],
+				[AC_MSG_RESULT(yes)],
+				[AC_MSG_RESULT(no)
+				 AC_MSG_ERROR([$1 provider was requested as direct, but is missing fi_direct.hand fi_direct.h.in])]
```

Comment:
Libfabric generally uses the Linux kernel style.  I do believe there's a script for checking that stuff... checking...  Yes, here's `checkpatch.pl`: https://github.com/torvalds/linux/blob/master/scripts/checkpatch.pl

You can hook it into your git pre-commit to prevent you from committing until it conforms to the right style, if you want.